IDE/ECLIPSE/SIFIVE/main.c

```diff
@@ -0,0 +1,115 @@
+/* main.c
+ *
+ * Copyright (C) 2019 wolfSSL Inc.
+ *
+ * This file is part of wolfSSL.
+ *
+ * wolfSSL is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfSSL is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1335, USA
+ */
+#include <wolfssl/wolfcrypt/settings.h>
+#include <wolfcrypt/test/test.h>
+#include <wolfcrypt/benchmark/benchmark.h>
+
+/* wolfCrypt_Init/wolfCrypt_Cleanup */
+#include <wolfssl/wolfcrypt/wc_port.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdint.h>
+
+#ifndef NO_CRYPT_BENCHMARK
+
+/*-specs=nano.specs doesn’t include support for floating point in printf()*/
+asm (".global _printf_float");
+
+#define RTC_FREQ    32768
+#define CLINT_MTIME_ADDR  0x200bff8
+#define WOLFSSL_SIFIVE_RISC_V_DEBUG 0
+
+double current_time(int reset)
+{
+    volatile uint64_t * mtime  = (uint64_t*) (CLINT_MTIME_ADDR);
+    uint64_t now = *mtime;
+    (void)reset;
+    return now/RTC_FREQ;
+}
+#endif
+
+void check(int depth) {
+    char ch;
+    char *ptr = malloc(1);
+
+    printf("stack at %p, heap at %p\n", &ch, ptr);
+    if (depth <= 0) 
+        return;
+    
+    check(depth-1);
```

Comment:
This appears to be debug code only enabled with `WOLFSSL_SIFIVE_RISC_V_DEBUG`. Can/should you have a corresponding free() here as the stack unwinds at line 58? Also consider wrapping this function in `#ifdef WOLFSSL_SIFIVE_RISC_V_DEBUG`.